File: lib/models/student.dart
class StudentModel {
  final String id;
  final String name;
  final String registerNumber;
  final String rollNumber;
  final String dateOfBirth;
  final String department;
  final String course;
  final String year;
  final String section;
  final String semester;
  final String college;
  final String location;
  final String email;
  final String phone;
  final double attendancePercentage;
  final String status;

  const StudentModel({
    required this.id,
    required this.name,
    required this.registerNumber,
    this.rollNumber = '',
    this.dateOfBirth = '',
    required this.department,
    required this.course,
    required this.year,
    required this.section,
    required this.semester,
    required this.college,
    required this.location,
    required this.email,
    required this.phone,
    required this.attendancePercentage,
    this.status = 'Active',
  });

  factory StudentModel.fromJson(Map<String, dynamic> json) {
    return StudentModel(
      id: json['id']?.toString() ?? '',
      name: json['name']?.toString() ?? '',
      registerNumber: json['register_number']?.toString() ?? json['registerNumber']?.toString() ?? '',
      rollNumber: json['roll_number']?.toString() ?? json['rollNumber']?.toString() ?? '',
      dateOfBirth: json['date_of_birth']?.toString() ?? json['dateOfBirth']?.toString() ?? '',
      department: json['department']?.toString() ?? 'Artificial Intelligence and Data Science',
      course: json['course']?.toString() ?? 'B.TECH',
      year: json['year']?.toString() ?? '4th Year',
      section: json['section']?.toString() ?? 'A',
      semester: json['semester']?.toString() ?? 'VII',
      college: json['college']?.toString() ?? 'Sengunthar Engineering College',
      location: json['location']?.toString() ?? 'Tiruchengode, Tamil Nadu',
      email: json['email']?.toString() ?? '',
      phone: json['phone']?.toString() ?? '',
      attendancePercentage: (json['attendance_percentage'] as num?)?.toDouble() ??
          (json['attendancePercentage'] as num?)?.toDouble() ??
          90.0,
      status: json['status']?.toString() ?? 'Active',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'register_number': registerNumber,
      'roll_number': rollNumber,
      'date_of_birth': dateOfBirth,
      'department': department,
      'course': course,
      'year': year,
      'section': section,
      'semester': semester,
      'college': college,
      'location': location,
      'email': email,
      'phone': phone,
      'attendance_percentage': attendancePercentage,
      'status': status,
    };
  }
}

File: backend/models/student.py
from sqlalchemy import Column, Integer, String, Float, DateTime, Date, ForeignKey
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from database import Base

class Student(Base):
    __tablename__ = "students"

    id = Column(String, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), unique=True, nullable=True)
    name = Column(String, nullable=False)
    register_number = Column(String, unique=True, index=True, nullable=False)
    roll_number = Column(String, default="")
    date_of_birth = Column(Date, nullable=True)
    department = Column(String, nullable=False, default="Artificial intelligence and Data science")
    department_id = Column(Integer, ForeignKey("departments.id"), nullable=True)
    course = Column(String, default="B.TECH")
    year = Column(String, default="IV")
    year_of_study = Column(Integer, default=4)
    section = Column(String, default="A")
    semester = Column(String, default="VII")
    college = Column(String, default="Sengunthar Engineering College")
    location = Column(String, default="Tiruchengode, Tamil Nadu")
    email = Column(String, unique=True, nullable=False)
    phone = Column(String, nullable=False)
    attendance_percentage = Column(Float, default=0.0)
    status = Column(String, default="Active")
    admission_year = Column(Integer, default=2023)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())

    user = relationship("User", back_populates="student_profile")
    dept = relationship("Department", back_populates="students")
